plan/plantest/rules.go

```diff
@@ -27,6 +27,70 @@ func (sr *SimpleRule) Name() string {
 	return "simple"
 }
 
+// SmashPlanRule adds an `Intruder` as predecessor of the given `Node` without
+// marking it as successor of it. It breaks the integrity of the plan.
+// If `Kind` is specified, it takes precedence over `Node`, and the rule will use it
+// to match.
+type SmashPlanRule struct {
+	Node plan.PlanNode
+	Intruder plan.PlanNode
+	Kind plan.ProcedureKind
+}
+
+func (SmashPlanRule) Name() string {
+	return "SmashPlanRule"
+}
+
+func (spp SmashPlanRule) Pattern() plan.Pattern {
+	var k plan.ProcedureKind
+	if len(spp.Kind) > 0 {
+		k = spp.Kind
+	} else {
+		k = spp.Node.Kind()
+	}
+
+	return plan.Pat(k, plan.Any())
+}
+
+func (spp SmashPlanRule) Rewrite(node plan.PlanNode) (plan.PlanNode, bool, error) {
+	if len(spp.Kind) > 0 || node == spp.Node {
+		node.AddPredecessors(spp.Intruder)
+	}
+	return node, false, nil
```

Comment:
It's worth noting in a comment here that you're returning `false` even though the plan has changed.